libs/net/dal/Services/NotificationService.cs

```diff
@@ -75,12 +124,60 @@ public override Notification Update(Notification entity)
             }
         });
 
+
+        // Add/Update the report template.
+        if (entity.Template != null)
```

Comment:
Notification updates will update the template.  Same thing happens with reports.  I don't love this solution, but it works for us currently.